### PR TITLE
4890 mixmax triggers and actions

### PR DESCRIPTION
**Pull request by luancazarine**
Mixmax triggers and actions

Triggers:
  - New Contact
  - New Contact Group

Actions:
  - Get Contacts
  - Create Contact
  - Update Contact
  - Search Contacts
  - Get Contact Groups
  - Create Contact Group

### CODE REVIEW

File: components/mixmax/sources/common/base-single.mjs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
```
Comment (feyzullah) [inline components/mixmax/sources/common/base-single.mjs:1]: Do we need this file here at all?

File: components/mixmax/package.json [hunk 2]
```diff
@@ -12,5 +12,8 @@
   "license": "MIT",
   "publishConfig": {
     "access": "public"
+  },
+  "dependencies": {
+    "@pipedream/platform": "^1.1.1"
```
Comment (feyzullah) [inline components/mixmax/package.json:17]: I guess pnpm will still use `1.2.1` but can we change it `1.2.1` as it is the latest version?
```suggestion
    "@pipedream/platform": "^1.2.1"
```

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -2837,6 +2840,14 @@ importers:
   components/teamdeck:
     specifiers: {}
 
+  components/teamgate:
```
Comment (feyzullah) [inline pnpm-lock.yaml:2843]: I guess `Teamgate`'s dependencies were also added here and it seems it uses `@pipedream/platform` version `0.10.0`. If possible can we omit these in this PR?

**Review by feyzullah — CHANGES_REQUESTED** [review]
Hi @luancazarine, I added a few comments. Can you please check?

**Review by feyzullah — DISMISSED** [review]
LGTM!

**Comment by DilanAthukorala** [comment]
@luancazarine Observing following issues,


1. Create Contact Action: Groups are not listed
image


2. Create Group Action: Contact Id's are not passed to MixMax
image


3. Get Groups action doesn't return Groups, it only returns contacts. 


4. Get Contacts Action throws error
image


5. Search Contacts Throws error when sending request without "q" String
image

**Comment by luancazarine** [comment]
Hello @DilanAthukorala, About the issues:

1. Create Contact Action: Groups are not listed
Did you create the groups? it was supposed to be working.
image

2. Create Group Action: Contact Id's are not passed to MixMax
Fixed!

3. Get Groups action doesn't return Groups, it only returns contacts.
Fixed!

4. Get Contacts Action throws error
Fixed! 

5. Search Contacts Throws error when sending request without "q" String
This error comes from the API so I made it required

**Comment by DilanAthukorala** [comment]
@luancazarine For get contact groups action - error throws when sending request without fields. 
image

**Comment by DilanAthukorala** [comment]
@luancazarine Issue still remains for Get contacts action - No results returned when groups are selected. Can you try to replicate this on your side as well please?

**Comment by luancazarine** [comment]
> @luancazarine Issue still remains for Get contacts action - No results returned when groups are selected. Can you try to replicate this on your side as well please?

@DilanAthukorala 
image

**Comment by DilanAthukorala** [comment]
> > @luancazarine Issue still remains for Get contacts action - No results returned when groups are selected. Can you try to replicate this on your side as well please?
> 
> @DilanAthukorala image

@luancazarine You have to select at least one group to observe the issue

**Comment by DilanAthukorala** [comment]
Verification completed, ready for release.

**Review by lcaresia — APPROVED** [review]
LGTM!

**Pull request merged**